app/views/search/index.html.erb

```diff
@@ -1,51 +1,53 @@
 <h1><%= t('views.search.title') %></h1>
-
-<div class="modal fade" id="qrscanner" tabindex="-1" aria-labelledby="qrScannerLabel" aria-hidden="true">
-    <div class="modal-dialog">
-        <div class="modal-content">
-            <div class="modal-header">
-                <h1 class="modal-title fs-5" id="qrScannerLabel"><%= t 'views.search.qr_scanner_title' %></h1>
-                <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
-            </div>
-            <div class="modal-body">
-                Your scanner goes here
-            </div>
-            <div class="modal-footer">
-                <button type="button" class="btn btn-secondary" data-bs-dismiss="modal"><%= t 'defaults.close' %></button>
+<div data-controller="qrscanner">
```

Comment:
Is it necessary that this div (that I think connects the view with an additional controller) is wrapped around the entire rest? Maybe it doesn't have to contain the search field?

Reply:
```suggestion
                </div>
            </div>
        </div>
        <%= text_field_tag :lastsearch, params[:lastsearch], class: "form-control invisible", value: @lastsearch %>
    <% end %>
</div> //this would be possible a possible end of the qrscanner div
    <% if @results %>
    <h2>Your search results:</h2>
    <div class="row row-cols-1 row-cols-md-4 g-3">
```
Every button that uses `data-action="click->qrscanner#..."` needs to be inside of the div. We could move the `</div>` up a little bit as suggested above...